From 00d8817ab207a9f60e94e87acf4f170155aecd48 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 6 Feb 2019 20:45:16 +0000
Subject: [PATCH] arm: dts: Change downstream vchiq compatible string

The new cache line size mechanism requires a different vchiq compatible
string on BCM2836 and BCM2837, but the downstream dts files didn't
inherit the upstream changes.

See: https://github.com/raspberrypi/linux/issues/2643

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm2708-rpi.dtsi | 2 +-
 arch/arm/boot/dts/bcm2709-rpi.dtsi | 5 +++++
 arch/arm/boot/dts/bcm2709.dtsi     | 2 +-
 arch/arm/boot/dts/bcm2710.dtsi     | 2 +-
 4 files changed, 8 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm/boot/dts/bcm2709-rpi.dtsi

--- a/arch/arm/boot/dts/bcm2708-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2708-rpi.dtsi
@@ -68,7 +68,7 @@
 			status = "disabled";
 		};
 
-		mailbox@7e00b840 {
+		vchiq: mailbox@7e00b840 {
 			compatible = "brcm,bcm2835-vchiq";
 			reg = <0x7e00b840 0x3c>;
 			interrupts = <0 2>;
--- /dev/null
+++ b/arch/arm/boot/dts/bcm2709-rpi.dtsi
@@ -0,0 +1,5 @@
+#include "bcm2708-rpi.dtsi"
+
+&vchiq {
+	compatible = "brcm,bcm2836-vchiq", "brcm,bcm2835-vchiq";
+};
--- a/arch/arm/boot/dts/bcm2709.dtsi
+++ b/arch/arm/boot/dts/bcm2709.dtsi
@@ -1,6 +1,6 @@
 #include "bcm2836.dtsi"
 #include "bcm270x.dtsi"
-#include "bcm2708-rpi.dtsi"
+#include "bcm2709-rpi.dtsi"
 
 / {
 	soc {
--- a/arch/arm/boot/dts/bcm2710.dtsi
+++ b/arch/arm/boot/dts/bcm2710.dtsi
@@ -1,6 +1,6 @@
 #include "bcm2837.dtsi"
 #include "bcm270x.dtsi"
-#include "bcm2708-rpi.dtsi"
+#include "bcm2709-rpi.dtsi"
 
 / {
 	compatible = "brcm,bcm2837", "brcm,bcm2836";
